test/test_gmtime.c

```diff
@@ -0,0 +1,228 @@
+/*
+ * Phoenix-RTOS
+ *
+ * libphoenix
+ *
+ * test/test_strftime.c
+ *
+ * Copyright 2020 Phoenix Systems
+ * Author: Marcin Brzykcy
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include "time.h"
+#include "stdio.h"
+#include "stdlib.h"
+
+#define NCOLS 9
+#define T1_LEN 50
+
+static const time_t input_vector[T1_LEN];
+static const int output_vector[T1_LEN][NCOLS];
+
+void tm_to_str(struct tm *t, char *buff){
+	sprintf(buff, "tm_sec %d, tm_min %d, tm_hour %d, tm_mday %d\n"
+			"tm_mon %d tm_year %d tm_wday %d tm_yday %d tm_isdst %d",
+			t->tm_sec, t->tm_min, t->tm_hour, t->tm_mday, t->tm_mon, t->tm_year,
+			t->tm_wday, t->tm_yday, t->tm_isdst);
+
+}
+
+struct tm create_tm(const int* input){
```

Comment:
Don't pass struct as retval. Pass struct tm * to the function instead. Also, this function is copied all over the files. Put in some _common_ or _utils_ file